### PR TITLE
Upgrade host applications

### PR DESCRIPTION
**Pull request by mpolden**

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/SystemApplication.java
Comment (freva) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/SystemApplication.java]: Why not let `all()` (maybe renamed) return `List<List<SystemApplcation>>` that will define both order (first list) and parallelism (second list) instead of having this method and enum order dictate that?

**Comment by freva** [comment]
`controller-server` tests fail for some reason :man_shrugging:

**Comment by mpolden** [comment]
> controller-server tests fail for some reason

They fail for me as well, but it's unrelated to this change. I think it's due to `Vtag.currentVersion` being populated with a different version number now, before it was always`6.1.0`.

**Comment by mpolden** [comment]
Never mind. It's this change, will fix.

**Review by freva — APPROVED** [review]
:+1: 

We are working on implementing upgrade of `host-admin`, but it's not out yet and we can't merge this until it is. 

I'll merge this once we are ready.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/SystemApplication.java
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/SystemApplication.java]: This means that the first three applications will upgrade in parallel, which is a bit confusing, given that only the first two are mentioned here. It is also based on how the code is used. 

I'd prefer a DAG representation, and a topological sort when deploying, if this ever grows in complexity.

**Comment by mpolden** [comment]
Ignored the host applications until they support upgrading.

@jonmv Please review last two commits and merge as I want the other changes in this PR.

**Review by freva — APPROVED** [review]

**Pull request merged**
